### ISSUE TITLE
`Get-NsxtServerDetail` returns `null`

### ISSUE THREAD

**Issue by nzRegularIT** [vmware/power-validated-solutions-for-cloud-foundation#286]
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

4.5.0

### Module Version

2.4.0.1008

### PowerShell Version

Windows PowerShell 5.1

### PowerCLI Version

13.1.0.21624340

### PowerVCF Version

2.3.0.1004

### Guest Operating System

Windows Server 2019

### Description

Run `Get-NsxtServerDetail` returns a null response when providing required parameter details.

### Error or Debug Output

see below

### Expected Behavior

Retrieves the FQDN, root and admin credentials of NSX for a given Workload Domain

### Actual Behavior

```powershell
Get-NsxtServerDetail `
	-fqdn "vcf-server.fqdn" `
	-username "user@vsphere.local"`
	-password "" `
	-domain "vcf-domain" `
	-listNodes
```

```
fqdn      :
adminUser :
adminPass :
rootUser  :
rootPass  :
nodes     :
```

```powershell
Get-NsxtServerDetail `
	-fqdn "vcf-server.fqdn" `
	-username "user@vsphere.local"`
	-password "" `
	-domaintype "MANAGEMENT"
```

```
fqdn      :
adminUser :
adminPass :
rootUser  :
rootPass  :
nodes     :
```

### Steps to Reproduce

Run `Request-VCFToken` to get token

Run `Get-NsxtServerDetail`, providing either domain name of domain type

### Possible resolution
Replace the below code, as Get-VCFNsxtcluster appears to return a null result unless the -id is specified. 

```powershell
$nsxtServerDetails = Get-VCFNsxtcluster | Where-Object { $_.id -eq $($vcfWorkloadDomainDetails.nsxtCluster.id) }
```

with
```powershell
$nsxtServerDetails = Get-VCFNsxtcluster -id $( $vcfWorkloadDomainDetails.nsxtCluster.id )
```

`Get-VCFNsxtCluster` does state that the id is optional, however I have not been able to get this to work without providing the id. https://vmware.github.io/powershell-module-for-vmware-cloud-foundation/documentation/functions/nsx/nsx-managers/Get-VCFNsxtCluster/

**Comment by tenthirtyam** [comment]
Hi @nzRegularIT - I'm not (yet) observing this error.

```powershell
➜ ssh vmware@x.x.x.x
(vmware@x.x.x.x) Password:
Last login: Thu Jul 13 23:31:24 2023 from x.x.x.x
 00:06:21 up 414 days, 11:26, 20 users,  load average: 0.01, 0.02, 0.00

3 Security notice(s)
Run 'tdnf updateinfo info' to see the details.

vmware@wdc330v-jb03 [ ~ ]$ pwsh
PowerShell 7.2.7
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

   A new PowerShell stable release is available: v7.3.6
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.3.6

PS /home/vmware> Install-Module -Name PowerValidatedSolutions

PS /home/vmware> Import-Module -Name PowerValidatedSolutions

PS /home/vmware> Get-NsxtServerDetail -fqdn sfo-vcf01.sfo.rainpole.io -username admin@local -password VMw@re1!VMw@re1! -domainType MANAGEMENT

fqdn      : sfo-m01-nsx01.sfo.rainpole.io
adminUser : admin
adminPass : o^6F@810zZiQNtyqJmwD
rootUser  : root
rootPass  : VMw@re1!VMw@re1!

PS /home/vmware> Get-NsxtServerDetail -fqdn sfo-vcf01.sfo.rainpole.io -username admin@local -password VMw@re1!VMw@re1! -domainType MANAGEMENT -listnodes

fqdn      : sfo-m01-nsx01.sfo.rainpole.io
adminUser : admin
adminPass : o^6F@810zZiNQtyqJmwD
rootUser  : root
rootPass  : VMw@re1!VMw@re1!
nodes     : {@{id=dc893e2b-ab66-4470-9959-d94f46319cb4; fqdn=sfo-m01-nsx01a.sfo.rainpole.io; ipAddress=172.18.95.72; name=sfo-m01-nsx01a}, @{id=65495e5d-5d0b-4b85-baeb-3134d54479a4; fqdn=sfo-m01-nsx01b.sfo.rainpole.io;
            ipAddress=172.18.95.73; name=sfo-m01-nsx01b}, @{id=db0636e9-1af9-4547-9534-1caacb5f0793; fqdn=sfo-m01-nsx01c.sfo.rainpole.io; ipAddress=172.18.95.74; name=sfo-m01-nsx01c}}

PS /home/vmware>
```

**Comment by tenthirtyam** [comment]
Hi Aaron (@nzRegularIT) - I wanted to follow-up and see if you are still consistently getting this error in one (or more) environments.
